backend/src/main/java/com/bakdata/conquery/io/mina/PipedJacksonProtocolFilter.java

```diff
@@ -0,0 +1,211 @@
+						break;
+					}
+
+					final int remaining = buffer.getInt();
+
+					asyncReader = newAsyncReader(remaining, nextFilter, session);
+					log.trace("Starting new reader for {}", DataSize.bytes(remaining));
+				}
+				else {
+					log.trace("Continuing with {}", DataSize.bytes(asyncReader.getRemaining()));
+				}
+
+				final boolean finished = asyncReader.receive(buffer);
+
+				if (finished) {
+					session.setAttribute(READER_KEY, null);
+				}
+			}
+		}
+	}
+
+	/**
+	 * If a prior edge-buffer exists, we need to join the incoming buffer with it.
+	 */
+	private IoBuffer handleExistingEdgeBuffer(IoSession session, IoBuffer buffer) throws IOException {
+		final IoBuffer edgeBuffer = (IoBuffer) session.getAttribute(EDGE_BUFFER);
+
+		if (edgeBuffer == null) {
+			return buffer;
+		}
+
+		log.debug("Found existing edge-buffer {}, appending {}", DataSize.bytes(edgeBuffer.position()), DataSize.bytes(buffer.limit()));
+
+		buffer.asInputStream().transferTo(edgeBuffer.asOutputStream());
+		buffer.flip();
+
+		session.setAttribute(EDGE_BUFFER, null);
+
+		return edgeBuffer;
+	}
+
+	private Reader getAsyncReader(IoSession session) {
+		return (Reader) session.getAttribute(READER_KEY);
+	}
+
+	/**
+	 * Test if we have enough contents to read the incoming buffer's length.
+	 * If not, cumulate the remaining buffer into a new buffer. The next incoming buffer will then be appended, ensuring enough content to read a length.
+	 */
+	private boolean requiresEdgeBuffering(IoSession session, IoBuffer buffer) throws IOException {
+		if (buffer.remaining() >= Integer.BYTES) {
+			return false;
+		}
+
+		final IoBuffer edgeBuffer = IoBuffer.allocate(64, false);
+		edgeBuffer.setAutoExpand(true);
+
+		buffer.asInputStream().transferTo(edgeBuffer.asOutputStream());
+
+		session.setAttribute(EDGE_BUFFER, edgeBuffer);
+
+		log.info("Not enough content in current buffer, resorting to edge-buffering.");
+		return true;
+	}
+
+	@NotNull
+	private Reader newAsyncReader(int remaining, NextFilter nextFilter, IoSession session) {
+		final Reader reader = new Reader(remaining, mapper, nextFilter, session);
+
+		readerThreads.submit(reader::read);
+
+		session.setAttribute(READER_KEY, reader);
+		return reader;
+	}
+
+	@Override
+	public synchronized void filterWrite(NextFilter nextFilter, IoSession session, WriteRequest writeRequest) throws Exception {
+		//TODO use CachedBufferAllocator?
+		final IoBuffer buf = IoBuffer.allocate(64, false);
+		buf.setAutoExpand(true);
+		buf.putInt(0); //DUMMY
+
+
+		final Stopwatch stopwatch = Stopwatch.createStarted();
+		log.trace("BEGIN Encoding message");
+
+		final Object message = writeRequest.getMessage();
+		try (OutputStream outputStream = buf.asOutputStream()) {
+			mapper.writerFor(NetworkMessage.class).writeValue(outputStream, message);
+		}
+
+		log.trace("FINISHED Encoding message in {}. Buffer size: {}. Message: {}", stopwatch, DataSize.bytes(buf.remaining()), message);
```

Comment:
Das hilft beim debugging
```suggestion
		log.trace("FINISHED Encoding message in {} for session {}. Buffer size: {}. Message: {}", stopwatch, session.getId(), DataSize.bytes(buf.remaining()), message);
```

Reply:
Session sollte doch schon im MDC sein?

Reply:
Ich glaube da steht aber noch nicht die Id drinne, was praktisch ist, wenn man lokal debuged